src/python/pants/engine/isolated_process.py

```diff
@@ -132,6 +131,19 @@ class FallibleExecuteProcessResult:
     stderr: bytes
     exit_code: int
     output_directory_digest: Digest
+
+
+@dataclass(frozen=True)
+class FallibleExecuteProcessResultWithPlatform:
```

Comment:
Do you think that we want `ExecuteProcessResultWithPlatform` so that rules can request that as an alternative to `FallibleExecuteProcessResultWithPlatform`?

Reply:
Possibly, but we can add that in another PR when we actually need it, this one is long enough already